### PR TITLE
Firebase setup

### PR DESCRIPTION
**Pull request by steveccable**
Sets up an initial configuration of the firebase app.  Ideally this should be worked into our Redux store for components to access it as needed, once we have Redux fully set up.

### CODE REVIEW

**Comment by steveccable** [comment]
@FuadBalashov @n-kumari @psteele2 FYI, added firebase to our project in a pretty simple commit.  Have a look when you can, then we can merge this in.

**Comment by steveccable** [comment]
@FuadBalashov @psteele2 @n-kumari Poke

File: src/index.js [hunk 1]
```diff
@@ -1,5 +1,19 @@
 import React from 'react';
 import MapScreen from './Components/MapScreen/MapScreen';
+import * as firebase from 'firebase';
```
Comment (n-kumari) [inline src/index.js:3]: Do we need to import * here?

File: package.json [hunk 2]
```diff
@@ -24,6 +24,8 @@
   },
   "dependencies": {
     "expo": "^20.0.0",
+    "firebase": "^4.6.1",
+    "immutable": "^3.8.2",
```
Comment (n-kumari) [inline package.json:28]: This is not being used now but we will use in future. Right?

Comment (steveccable) [reply package.json:28]: Yes, I would be pretty surprised if we built an entire RN app without using Immutable at some point.

File: src/index.js (hunk 1 above)
Comment (steveccable) [reply src/index.js:3]: Looks like most sources just use `import firebase from 'firebase'` so I'll give that a shot instead.

**Review by FuadBalashov — APPROVED** [review]
lgtm beyond Nisha's one comment

File: src/index.js (hunk 1 above)
Comment (psteele2) [reply src/index.js:3]: Merging with the understanding that Cable will fix this. This is so we can all base off of the meat of the PR.

**Pull request merged**